src/CommunityToolkit.Maui/Extensions/OnScreenSizeExtension.shared.cs

```diff
@@ -0,0 +1,146 @@
+{
+	static readonly object defaultNull = new ();
+
+	 Dictionary<ScreenCategories, object> categoryPropertyValues = new () {
+		{ ScreenCategories.ExtraSmall, defaultNull},
+		{ ScreenCategories.Small, defaultNull},
+		{ ScreenCategories.Medium,  defaultNull},
+		{ ScreenCategories.Large,  defaultNull},
+		{ ScreenCategories.ExtraLarge,  defaultNull},
+	};
+
+	/// <summary>
+	/// Xaml internal usage
+	/// </summary>
+	public OnScreenSizeExtension()
+	{
+		Default = defaultNull;
+	}
+
+	/// <summary>
+	/// Default value assumed when the other property values were not provided for the current device. 
+	/// </summary>
+	public object Default { get; set; }
+
+
+	/// <summary>
+	/// Defines a value used when the screen size is categorized as <see cref="ScreenCategories.ExtraSmall"/>
+	/// </summary>
+	public object ExtraSmall
+	{
+		get => categoryPropertyValues[ScreenCategories.ExtraSmall];
+		set => categoryPropertyValues[ScreenCategories.ExtraSmall] = value;
+	}
+	/// <summary>
+	/// Defines a value used when the screen size is categorized as <see cref="ScreenCategories.Small"/>
+	/// </summary>
+	public object Small
+	{
+		get => categoryPropertyValues[ScreenCategories.Small];
+		set => categoryPropertyValues[ScreenCategories.Small] = value;
+	}
+	
+	/// <summary>
+	/// Defines a value used when the screen size is categorized as <see cref="ScreenCategories.Medium"/>
+	/// </summary>
+	public object Medium
+	{
+		get => categoryPropertyValues[ScreenCategories.Medium];
+		set => categoryPropertyValues[ScreenCategories.Medium] = value;
+	}
+
+	/// <summary>
+	/// Defines a value used when the screen size is categorized as <see cref="ScreenCategories.Large"/>
+	/// </summary>
+	public object Large
+	{
+		get => categoryPropertyValues[ScreenCategories.Large];
+		set => categoryPropertyValues[ScreenCategories.Large] = value;
+	}
+
+	/// <summary>
+	/// Defines a value used when the screen size is categorized as <see cref="ScreenCategories.ExtraLarge"/>
+	/// </summary>
+	public object ExtraLarge
+	{
+		get => categoryPropertyValues[ScreenCategories.ExtraLarge];
+		set => categoryPropertyValues[ScreenCategories.ExtraLarge] = value;
+	}
+
+
+	/// <summary>
+	/// Xaml internal usage
+	/// </summary>
+	public object ProvideValue(IServiceProvider serviceProvider)
+	{
+		var valueProvider = serviceProvider?.GetService<IProvideValueTarget>() ?? throw new ArgumentException($"Service provided for OnScreenSize is null");
+
+		BindableProperty bp;
+		PropertyInfo pi = null!;
+		
+		if (valueProvider.TargetObject is Setter setter)
+		{
+			bp = setter.Property;
+		}
+		else
+		{
+			bp = (valueProvider.TargetProperty as BindableProperty)!;
+			pi = (valueProvider.TargetProperty as PropertyInfo)!;
+		}
+
+		LogHelpers.Log($"Providing Value using propertyType:\"{(bp?.ReturnType ?? pi?.PropertyType ?? null)}\" and BindableProperty:{(bp ?? null)}", LogLevel.Debug);
+
+		var propertyType = bp?.ReturnType ?? pi?.PropertyType;
+
+		if (propertyType == null)
```

Comment:
We don't use `==` or `!=` to check null, please use `is` or `is not` respectively 
```suggestion
		if (propertyType is null)
```